find-unused-vars.coffee

```diff
@@ -0,0 +1,163 @@
+
+      if node.arguments.length > 0
+        for arg in node.arguments
+          if checkIdentifier arg
+            addStatsEntry arg.name
+            variablesStats[arg.name].calls += 1
+
+    # variable = variable2 -- to find variable2 used
+    else if node.type is nodeType.AssignmentExpression and  checkIdentifier node.right
+      addStatsEntry node.right.name
+      variablesStats[node.right.name].calls += 1
+
+    # variable = this.variable2 -- to find variable2 used
+    else if node.type is nodeType.AssignmentExpression and node.right.type is nodeType.MemberExpression
+      if node.right.object.type
+        addStatsEntry node.right.object.name
+        variablesStats[node.right.object.name].calls += 1
+
+    # variable
+    else if node.type is nodeType.VariableDeclarator
+      addStatsEntry node.id.name
+      variablesStats[node.id.name].declarations += 1
+
+    # 'test' : variable
+    else if node.type is nodeType.Property
+      if checkIdentifier node.value
+        addStatsEntry node.value.name
+        variablesStats[node.value.name].calls += 1
+
+    #  variable1 <= variable2
+    else if node.type is nodeType.ConditionalExpression
+      if checkIdentifier node.consequent
+        addStatsEntry node.consequent.name
+        variablesStats[node.consequent.name].calls += 1
+
+    # variable2 = someVariable.variable1 - variable1 is used
+    else if node.type is nodeType.MemberExpression and node.property?
+      if checkIdentifier node.object
+        addStatsEntry node.object.name
+        variablesStats[node.object.name].calls += 1
+
+    # if (variable)
+    else if node.type is nodeType.IfStatement
+      if checkIdentifier node.test
+        addStatsEntry node.test.name
+        variablesStats[node.test.name].calls += 1
+
+      if node.test.type is nodeType.UnaryExpression
+        addStatsEntry node.test.argument.name
+        variablesStats[node.test.argument.name].calls += 1
+
+    # variable1 || variable2 or variable1 && variable2
+    else if node.type is nodeType.LogicalExpression
+      if checkIdentifier node.left
+        addStatsEntry node.left.name
+        variablesStats[node.left.name].calls += 1
+
+      if checkIdentifier node.right
+        addStatsEntry node.right.name
+        variablesStats[node.right.name].calls += 1
+
+    # comparision two variable variable1 != null or variable1 != variable2
+    else if node.type is nodeType.BinaryExpression
+      if checkIdentifier node.left
+        addStatsEntry(node.left.name)
+        variablesStats[node.left.name].calls += 1
+
+      if checkIdentifier node.right
+        addStatsEntry node.right.name
+        variablesStats[node.right.name].calls += 1
+
+    # return variable
+    else if node.type is nodeType.ReturnStatement
+      if node.argument? and checkIdentifier node.argument
+        addStatsEntry node.argument.name
+        variablesStats[node.argument.name].calls += 1
+
+  processResults variablesStats
```

Comment:
for the sake of extensibility, do not call processResults inside of this function.